### DISCUSSION TITLE
[REQ] AI audio libraries "port"

### DISCUSSION THREAD

**Discussion by MarcoRavich** [intel/openvino-plugins-ai-audacity#292] [Ideas]
Hi there,
many interesting projects are growing around "AI" audio, it would be great to have an OpenVINO version too:
- @asteroid-team's [torch-audiomentations](https://github.com/asteroid-team/torch-audiomentations#readme) - Audio data augmentation in PyTorch, inspired by [audiomentations](https://github.com/iver56/audiomentations#readme);
- @iver56's [audiomentations](https://github.com/iver56/audiomentations#readme) - A Python library for audio data augmentation, inspired by [albumentations](https://github.com/albu/albumentations):
  #### Transforms:
    - [AddBackgroundNoise](https://iver56.github.io/audiomentations/waveform_transforms/add_background_noise/): Mixes in another sound to add background noise
    - [AddColorNoise](https://iver56.github.io/audiomentations/waveform_transforms/add_color_noise/): Adds noise with specific color
    - [AddGaussianNoise](https://iver56.github.io/audiomentations/waveform_transforms/add_gaussian_noise/): Adds gaussian noise to the audio samples
    - [AddGaussianSNR](https://iver56.github.io/audiomentations/waveform_transforms/add_gaussian_snr/): Injects gaussian noise using a randomly chosen signal-to-noise ratio
    - [AddShortNoises](https://iver56.github.io/audiomentations/waveform_transforms/add_short_noises/): Mixes in various short noise sounds
    - [AdjustDuration](https://iver56.github.io/audiomentations/waveform_transforms/adjust_duration/): Trims or pads the audio to fit a target duration
    - [AirAbsorption](https://iver56.github.io/audiomentations/waveform_transforms/air_absorption/): Applies frequency-dependent attenuation simulating air absorption
    - [Aliasing](https://iver56.github.io/audiomentations/waveform_transforms/aliasing/): Produces aliasing artifacts by downsampling without low-pass filtering and then upsampling
    - [ApplyImpulseResponse](https://iver56.github.io/audiomentations/waveform_transforms/apply_impulse_response/): Convolves the audio with a randomly chosen impulse response
    - [BandPassFilter](https://iver56.github.io/audiomentations/waveform_transforms/band_pass_filter/): Applies band-pass filtering within randomized parameters
    - [BandStopFilter](https://iver56.github.io/audiomentations/waveform_transforms/band_stop_filter/): Applies band-stop (notch) filtering within randomized parameters
    - [BitCrush](https://iver56.github.io/audiomentations/waveform_transforms/bit_crush/): Applies bit reduction without dithering
    - [Clip](https://iver56.github.io/audiomentations/waveform_transforms/clip/): Clips audio samples to specified minimum and maximum values
    - [ClippingDistortion](https://iver56.github.io/audiomentations/waveform_transforms/clipping_distortion/): Distorts the signal by clipping a random percentage of samples
    - [Gain](https://iver56.github.io/audiomentations/waveform_transforms/gain/): Multiplies the audio by a random gain factor
    - [GainTransition](https://iver56.github.io/audiomentations/waveform_transforms/gain_transition/): Gradually changes the gain over a random time span
    - [HighPassFilter](https://iver56.github.io/audiomentations/waveform_transforms/high_pass_filter/): Applies high-pass filtering within randomized parameters
    - [HighShelfFilter](https://iver56.github.io/audiomentations/waveform_transforms/high_shelf_filter/): Applies a high shelf filter with randomized parameters
    - [Lambda](https://iver56.github.io/audiomentations/waveform_transforms/lambda/): Applies a user-defined transform
    - [Limiter](https://iver56.github.io/audiomentations/waveform_transforms/limiter/): Applies dynamic range compression limiting the audio signal
    - [LoudnessNormalization](https://iver56.github.io/audiomentations/waveform_transforms/loudness_normalization/): Applies gain to match a target loudness
    - [LowPassFilter](https://iver56.github.io/audiomentations/waveform_transforms/low_pass_filter/): Applies low-pass filtering within randomized parameters
    - [LowShelfFilter](https://iver56.github.io/audiomentations/waveform_transforms/low_shelf_filter/): Applies a low shelf filter with randomized parameters
    - [Mp3Compression](https://iver56.github.io/audiomentations/waveform_transforms/mp3_compression/): Compresses the audio to lower the quality
    - [Normalize](https://iver56.github.io/audiomentations/waveform_transforms/normalize/): Applies gain so that the highest signal level becomes 0 dBFS
    - [Padding](https://iver56.github.io/audiomentations/waveform_transforms/padding/): Replaces a random part of the beginning or end with padding
    - [PeakingFilter](https://iver56.github.io/audiomentations/waveform_transforms/peaking_filter/): Applies a peaking filter with randomized parameters
    - [PitchShift](https://iver56.github.io/audiomentations/waveform_transforms/pitch_shift/): Shifts the pitch up or down without changing the tempo
    - [PolarityInversion](https://iver56.github.io/audiomentations/waveform_transforms/polarity_inversion/): Flips the audio samples upside down, reversing their polarity
    - [RepeatPart](https://iver56.github.io/audiomentations/waveform_transforms/repeat_part/): Repeats a subsection of the audio a number of times
    - [Resample](https://iver56.github.io/audiomentations/waveform_transforms/resample/): Resamples the signal to a randomly chosen sampling rate
    - [Reverse](https://iver56.github.io/audiomentations/waveform_transforms/reverse/): Reverses the audio along its time axis
    - [RoomSimulator](https://iver56.github.io/audiomentations/waveform_transforms/room_simulator/): Simulates the effect of a room on an audio source
    - [SevenBandParametricEQ](https://iver56.github.io/audiomentations/waveform_transforms/seven_band_parametric_eq/): Adjusts the volume of 7 frequency bands
    - [Shift](https://iver56.github.io/audiomentations/waveform_transforms/shift/): Shifts the samples forwards or backwards
    - [SpecChannelShuffle](https://iver56.github.io/audiomentations/spectrogram_transforms/): Shuffles channels in the spectrogram
    - [SpecFrequencyMask](https://iver56.github.io/audiomentations/spectrogram_transforms/): Applies a frequency mask to the spectrogram
    - [TanhDistortion](https://iver56.github.io/audiomentations/waveform_transforms/tanh_distortion/): Applies tanh distortion to distort the signal
    - [TimeMask](https://iver56.github.io/audiomentations/waveform_transforms/time_mask/): Makes a random part of the audio silent
    - [TimeStretch](https://iver56.github.io/audiomentations/waveform_transforms/time_stretch/): Changes the speed without changing the pitch
    - [Trim](https://iver56.github.io/audiomentations/waveform_transforms/trim/): Trims leading and trailing silence from the audio
- @FedericoDiMarzo's [torch_utilities](https://github.com/FedericoDiMarzo/torch_utilities#readme) - a collection of tools and utilities for working with PyTorch in the audio domain
- @ritik8801's [Audio Data Analysis Using Deep Learning with Python](https://github.com/ritik8801/Audio-Data-Analysis-Using-Deep-Learning-with-Python#readme) - audio data analysis and extracted necessary features from a sound/audio file

More to come.

**Comment by MarcoRavich** [comment]
- @ckx2001's [AudioCraft](https://github.com/ckx2001/audio_enhance#readme):
  - [MusicGen](https://github.com/ckx2001/audio_enhance/blob/main/docs/MUSICGEN.md): A state-of-the-art controllable text-to-music model.
  - [AudioGen](https://github.com/ckx2001/audio_enhance/blob/main/docs/AUDIOGEN.md): A state-of-the-art text-to-sound model.
  - [EnCodec](https://github.com/ckx2001/audio_enhance/blob/main/docs/ENCODEC.md): A state-of-the-art high fidelity neural audio codec.
  - [Multi Band Diffusion](https://github.com/ckx2001/audio_enhance/blob/main/docs/MBD.md): An EnCodec compatible decoder using diffusion.
  - [MAGNeT](https://github.com/ckx2001/audio_enhance/blob/main/docs/MAGNET.md): A state-of-the-art non-autoregressive model for text-to-music and text-to-sound.
  - [AudioSeal](https://github.com/ckx2001/audio_enhance/blob/main/docs/WATERMARKING.md): A state-of-the-art audio watermarking.

**Reply by RyanMetcalfeInt8** [reply]
Just to point out, the Music Generation feature in these plugjns is currently based on this MusicGen model (and uses EnCodec also for conversions).

**Reply by RyanMetcalfeInt8** [reply]
btw, it's not really clear what the difference is between @ckx2001's repo, and the official AudioCraft repo: https://github.com/facebookresearch/audiocraft

**Reply by MarcoRavich** [reply]
> btw, it's not really clear what the difference is between @ckx2001's repo, and the official AudioCraft repo: https://github.com/facebookresearch/audiocraft

I honestly don't know since it don't reports as a fork... bus seems pretty fresh committed so maybe a mod ? 

While @ckx2001 cuold certainly explain better, the @facebookresearch's one is the obvious way to go.

**Comment by MarcoRavich** [comment]
Forgot to mention @HENDRIX-ZT2's [Python Audio Restoration Suite](https://github.com/HENDRIX-ZT2/pyaudiorestoration#readme) which implements very useful (even if "non-AI"):

- Wow & Flutter Removal
- Speed matching to hum frequency
- EQ matching with differential EQ
- Sub-sample accurate Spectral Temporal Alignment
- Automatic Dropout Restoration
- Spectral Expander / Decompressor

Hope that inspires !

**Comment by MarcoRavich** [comment]
Bump.

Just discovered the cool @mikeoliphant's [NeuralAudio library](https://github.com/mikeoliphant/NeuralAudio#readme), _a C++ library designed to make it easy to use neural network machine learning models (ie: guitar amplifier captures/profiles) in real-time audio applications._
